airflow/executors/executor_loader.py

```diff
@@ -41,6 +41,22 @@
     from airflow.executors.base_executor import BaseExecutor
 
 
+def _validate_database_executor_compatibility(executor: type[BaseExecutor]) -> None:
```

Comment:
do we want to write tests for this function?

Reply:
Added